### PR TITLE
remove unused streaming context wrappers

### PR DESCRIPTION
**Pull request by camdencheek**
This removes a bunch of dead context code. Tagging search core because it
looks like this was once useful, but now we're not using it and I'm not sure if we
should be. 

## Test plan

This just removes dead code.

### CODE REVIEW

File: internal/search/streaming/stream.go [hunk 1]
```diff
@@ -72,14 +72,9 @@ func (s *LimitStream) Send(event SearchEvent) {
 // Stream are complete.
 func WithLimit(ctx context.Context, parent Sender, limit int) (context.Context, Sender, context.CancelFunc) {
 	ctx, cancel := context.WithCancel(ctx)
-	mutCtx := WithMutableValue(ctx)
-	cancelWithReason := func() {
-		mutCtx.Set(CanceledLimitHit, true)
-		cancel()
-	}
-	stream := &LimitStream{cancel: cancelWithReason, s: parent}
```
Comment (camdencheek) [inline internal/search/streaming/stream.go:80]: This is the only place we used any of the mutable context stuff, and we were never reading the context value outside of tests.

**Review by rvantonder — APPROVED** [review]
Stamping to unblock, feel free to wait for core approval.

**Review by keegancsmith — APPROVED** [review]
LGTM. We debated if we should add this code since its a bit miffy to add this to a context, but it solved a real issue at the time. Glad we no longer need it :)

cc @stefanhengl who added this.

I think we lost the need for this when @tsenart introduced paginated repo resolution.

File: internal/search/streaming/stream.go (hunk 1 above)
Comment (stefanhengl) [reply internal/search/streaming/stream.go:80]: Thanks for removing this! I think this is a left over from before Tomas introduced the repo iterator.

**Review by stefanhengl — APPROVED** [review]

**Pull request merged**